### ISSUE TITLE
Custom Styles not working

### ISSUE THREAD

**Issue by michael-lianto** [stephy/CalendarPicker#117]
Hi, i'm trying to implement custom style to each date (i need to mark the date which got an event in it). tried the example in the readme, but it didn't work (no custom styles applied, just a normal looking calendar showed).

i'm using 
```
"react": "16.0.0-alpha.12",
"react-native-calendar-picker": "^5.11.0",
"react-native": "https://github.com/expo/react-native/archive/sdk-21.0.2.tar.gz",
```

and my code
```
constructor(props){
		super(props);
		this.state = {
			date: new Date(),
			schedule: [],
			dateStyle: []
		}
	}

componentWillMount(){
		let today = moment();
		let day = today.clone();
		let customDatesStyles = [];
		while(day.add(1, 'day').isSame(today, 'month')) {
			customDatesStyles.push({
				date: day.clone(),
				// Random colors
				style: {backgroundColor: '#'+('#00000'+(Math.random()* 1<<24)|0).toString(16)).slice(-6)},
			});
		}

		this.setState({dateStyle: customDatesStyles});

	}
render() {
  	var el = this;

	  console.log(this.state.dateStyle);
    return (
      <Container style={{ backgroundColor: '#fff' }}>
				<Content style={{ backgroundColor: 'transparent' }}>
				
						<CalendarPicker
						selectedDate={this.state.date}
						onDateChange={this.onDateChange.bind(this)}
						customDatesStyles={this.state.dateStyle}

						/>
				</Content>
      </Container>

```

the console log result is this
```
Array [
  Object {
    "date": "2018-04-12T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#76c717",
    },
  },
  Object {
    "date": "2018-04-13T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#6a049c",
    },
  },
  Object {
    "date": "2018-04-14T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#59792c",
    },
  },
  Object {
    "date": "2018-04-15T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#430949",
    },
  },
  Object {
    "date": "2018-04-16T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#a2787c",
    },
  },
  Object {
    "date": "2018-04-17T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#77d857",
    },
  },
  Object {
    "date": "2018-04-18T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#cb5d85",
    },
  },
  Object {
    "date": "2018-04-19T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#c4065d",
    },
  },
  Object {
    "date": "2018-04-20T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#60c7b4",
    },
  },
  Object {
    "date": "2018-04-21T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#182905",
    },
  },
  Object {
    "date": "2018-04-22T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#819dc3",
    },
  },
  Object {
    "date": "2018-04-23T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#c074ef",
    },
  },
  Object {
    "date": "2018-04-24T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#ad6e1d",
    },
  },
  Object {
    "date": "2018-04-25T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#894728",
    },
  },
  Object {
    "date": "2018-04-26T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#1d872d",
    },
  },
  Object {
    "date": "2018-04-27T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#3fdeef",
    },
  },
  Object {
    "date": "2018-04-28T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#41c701",
    },
  },
  Object {
    "date": "2018-04-29T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#95b705",
    },
  },
]
```

the result is just a normal calendar without any changes in background color, despite the example screenshot looks different than mine, 

How am i supposed to implement the custom style (different style for each date)?

**Comment by peacechen** [comment]
The feature to custom style dates was published in 5.12.0.  Check which version is actually installed.  I'd expect it to be newer than 5.11.0 since the semver is prefixed by ^.

As a test, try installing version 5.12.0 specifically.

**Comment by peacechen** [comment]
The latest release, 5.18.0, removed all the unnecessary expo dependencies.  If you're not using expo, I recommend updating to 5.18.0.  You can then remove dependencies such as babel-preset-expo, etc

**Comment by michael-lianto** [comment]
i have tried 5.12.0, but the custom style still not showing up. I'm using Expo SDK 21.
here's my package.json
```
{
  "name": "project",
  "version": "1.0.0",
  "private": true,
  "devDependencies": {
    "babel-eslint": "^6.1.2",
    "babel-plugin-transform-async-to-generator": "^6.24.1",
    "chai": "^3.5.0",
    "eslint": "^3.5.0",
    "eslint-config-airbnb": "^11.1.0",
    "eslint-plugin-import": "^1.14.0",
    "eslint-plugin-jsx-a11y": "^2.2.1",
    "eslint-plugin-react": "^6.2.0",
    "eslint-plugin-react-native": "^2.0.0",
    "jest-expo": "^21.0.0",
    "mocha": "^2.5.3",
    "react-native-scripts": "0.0.26",
    "react-test-renderer": "~15.4.1"
  },
  "main": "./node_modules/react-native-scripts/build/bin/crna-entry.js",
  "scripts": {
    "postinstall": "remotedev-debugger",
    "start": "react-native-scripts start",
    "eject": "react-native-scripts eject",
    "android": "react-native-scripts android",
    "ios": "react-native-scripts ios",
    "eslint": "eslint",
    "test": "node node_modules/jest/bin/jest.js --watch"
  },
  "upstreamRepo": "git@gitstrap.com:strapmobile/NativeStarterPro-seed.git",
  "jest": {
    "preset": "jest-expo"
  },
  "dependencies": {
    "@expo/vector-icons": "^5.0.0",
    "babel-preset-expo": "^4.0.0",
    "color": "^0.11.3",
    "expo": "^21.0.0",
    "i18next": "^10.5.0",
    "lodash": "^4.13.1",
    "moment": "^2.21.0",
    "native-base": "2.3.2",
    "react": "16.0.0-alpha.12",
    "react-i18next": "^7.5.0",
    "react-native": "https://github.com/expo/react-native/archive/sdk-21.0.2.tar.gz",
    "react-native-button": "^1.6.0",
    "react-native-calendar-picker": "^5.12.0",
    "react-native-easy-grid": "0.1.15",
    "react-native-extended-stylesheet": "^0.6.0",
    "react-native-gifted-messenger": "^0.1.4",
    "react-native-gifted-spinner": "0.0.4",
    "react-native-htmlview": "^0.12.1",
    "react-native-loading-spinner-overlay": "^0.5.1",
    "react-native-modalbox": "^1.3.7",
    "react-native-pinch-zoom-view": "^0.1.3",
    "react-native-popup-menu": "^0.8.3",
    "react-native-router-flux": "3.38.0",
    "react-native-scrollable-tab-view": "0.7.2",
    "react-native-simple-radio-button": "^2.6.2",
    "react-native-storage": "^0.2.2",
    "react-native-swiper": "1.5.4",
    "react-native-transformable-image": "0.0.18",
    "react-redux": "^4.4.5",
    "redux": "^3.5.2",
    "redux-persist": "^3.2.2",
    "redux-thunk": "^2.1.0",
    "remote-redux-devtools": "^0.3.3",
    "remote-redux-devtools-on-debugger": "^0.4.6"
  }
}

```

am i missing anything? or should i upgrade to the latest ver?

**Comment by michael-lianto** [comment]
Sorry for the late reply,

that was my fault, i tried with the example code, and i check the custom styles before passing it to the props, it shows similar with my previous code though

```
Array [
  Object {
    "date": "2018-04-12T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#76c717",
    },
  },
  Object {
    "date": "2018-04-13T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#6a049c",
    },
  },
  Object {
    "date": "2018-04-14T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#59792c",
    },
  },
  Object {
    "date": "2018-04-15T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#430949",
    },
  },
  Object {
    "date": "2018-04-16T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#a2787c",
    },
  },
  Object {
    "date": "2018-04-17T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#77d857",
    },
  },
  Object {
    "date": "2018-04-18T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#cb5d85",
    },
  },
  Object {
    "date": "2018-04-19T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#c4065d",
    },
  },
  Object {
    "date": "2018-04-20T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#60c7b4",
    },
  },
  Object {
    "date": "2018-04-21T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#182905",
    },
  },
  Object {
    "date": "2018-04-22T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#819dc3",
    },
  },
  Object {
    "date": "2018-04-23T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#c074ef",
    },
  },
  Object {
    "date": "2018-04-24T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#ad6e1d",
    },
  },
  Object {
    "date": "2018-04-25T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#894728",
    },
  },
  Object {
    "date": "2018-04-26T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#1d872d",
    },
  },
  Object {
    "date": "2018-04-27T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#3fdeef",
    },
  },
  Object {
    "date": "2018-04-28T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#41c701",
    },
  },
  Object {
    "date": "2018-04-29T18:39:23.284Z",
    "style": Object {
      "backgroundColor": "#95b705",
    },
  },
]
```
and ... it still didnt show the custom styled one

**Comment by peacechen** [comment]
Custom styles are working on multiple projects.  At this point I recommend that you create a Snack so we can diagnose the issue.
https://blog.expo.io/sketch-a-playground-for-react-native-16b2401f44a2

**Comment by michael-lianto** [comment]
Hi, thanks for your the snack that you made :)

I tried to copy the code directly, and i noticed i got this error,
undefined is not an object (evaluating 'this.props.selectedDate.getDate')
it's kinda weird because i am using the same react-native-calendar-picker as you did in the snack (5.18.0) and i checked the package json inside the node_modules, and confirmed it.
but in your snack, i dont find any selectedDate get initiated.

is this related to version issue?

note:
i am using yarn to install.
yarn add react-native-calendar-picker@5.18.0

**Comment by peacechen** [comment]
That snack doesn't pass in a `selectedDate` prop, so it's expected to be undefined.

Are the custom styles working for your app now?

**Comment by ithieund** [comment]
Hi @peacechen ,
Your snack worked perfectly when running standalone.
But when I try to install your module with the same code in a project using native-base, it's not working at all.

Maybe some code conflict with native-base themes???!!!!
You can download sample native-base app and try to make your module work here.
https://github.com/start-react/native-starter-kit

Can you please try and fix this?
Thanks.

**Comment by peacechen** [comment]
Would you create a snack with NativeBase?  I haven't used that before and am not familiar with how it changes the environment.  If NativeBase sets global component props, that could conflict with this library (though it would cause problems with lots of other libraries as well).

**Comment by ithieund** [comment]
@peacechen ,
You can try it out by pulling the native start kit project and then install your module after that.
I'm not farmiliar with using Snack so it's hard for me to create a snack with Native-Base.

I'm appreciate if you could figure out this issue.
Thank you very much.

**Comment by peacechen** [comment]
Sorry, have been very busy with deadlines here.  Would you post an example repo with Native Base, CalendarPicker and custom date styles?  If I find a fix, I can push the changes for you to look at.

Another debug option is to inspect the dates with the simulator and see what styles it's getting.  Maybe NativeBase is overriding them.

**Comment by peacechen** [comment]
Closing due to inactivity.  Feel free to re-open if you have more information.

**Comment by jordlay** [comment]
Hi,
Experienced this issue in 2022, I can see this was closed without a solution. I managed to solve this issue so thought I would explain how:
My code looked very much as above and it was not updating, so I used useState and useEffect hooks to force the update in real time

 const [customDatesStyles, setCustomDatesStyles] = useState([]);
  useEffect(()=>{
  }, [customDatesStyles])

The only change now is to create a temporary styles array (called temp here) that you push to and then use setCustomDatesStyles(temp).

Hope this helps anyone still stuck on this!
